### ISSUE TITLE
How to update the pdf background for a specific page? 

### ISSUE THREAD

**Issue by ez4lionky** [xournalpp/xournalpp#4337]
**Is your feature request related to a problem? Please describe.**
If I wrote some notes before the class for preview. But the tutor updates or adds some pages during the class, and I don't want to discard those notations. So I would like to manually add those pages or amend the pdf background for specific pages.
What should I do to implement the goal?

**Describe the solution you'd like**
Is it possible to add a function that we can manually replace the pdf background with a specific page of an external pdf file?

**Describe alternatives you've considered**
Or I just replace the original pdf with the new one, but the consequence is that my notation will be inconsistent with the new pdf, right?

**Comment by ez4lionky** [comment]
Thanks for your reply!

I understand that I can replace the previous pdf with the new one to update the background if the amendation will not affect other pages.

But it will be hairy if the tutor adds some pages between some pages. It means the annotations will be messed up due to extra pages, right?

The alternative method I adopted to avoid such a problem is manually adding a new page in an appropriate place. Then take a screenshot of the new page and paste it so I can take extra annotations. 

But if there is a more efficient way to do so, the user experience will be improved greatly (the screenshot is not as clear as the original pdf when enlarged to fit the page size).

**Comment by rolandlo** [comment]
Yes, if some pages between existing pages are inserted, then the annotations will be messed up. It would be possible to add those pages manually by editing the decompressed `.xopp`-file. That would still be preferable over taking screenshots. What is your OS?

**Comment by ez4lionky** [comment]
I usually worked in Ubuntu 20.04, so how to decompress the .xopp-file and insert new pdf pages?

**Comment by rolandlo** [comment]
About how to decompress the .xopp-file check https://github.com/xournalpp/xournalpp/issues/3189. 
The decompressed xopp-file will have sections like:
```xml
<page width="612" height="792">
<background type="pdf" pageno="5"/>
<layer/>
</page>
```
with possible some more content in between. The example is for PDF page 5 of width 612 pt and height 792 pt and no annotations.
You can add one or more of these sections (with the correct numbers for width, height and pageno) and increase all page numbers (in `pageno=`) that follow.

**Comment by ez4lionky** [comment]
Thanks for your information!
I will try to write a script to handle this problem. 
But look forward to a new UI that can add such a function.

**Comment by RichardScharf** [comment]
There is a workaround for this.
It is possible to render a single .pdf page at a time using the LaTeX tool with `\usepackage{pdfpages}` in the preamble.

then you can import it with `\includegraphics[page=3, size=0.8]{FILEPATH}` and it will persist after rendering

the size option is used to prevent Xournal++ from failing to render the right side of big LaTeX inputs.

**Comment by ez4lionky** [comment]
> There is a workaround for this. It is possible to render a single .pdf page at a time using the LaTeX tool with `\usepackage{pdfpages}` in the preamble.
> 
> then you can import it with `\includegraphics[page=3, size=0.8]{FILEPATH}` and it will persist after rendering
> 
> the size option is used to prevent Xournal++ from failing to render the right side of big LaTeX inputs.

It sounds like is a more efficient workaround, and can be integrated with the current system.
